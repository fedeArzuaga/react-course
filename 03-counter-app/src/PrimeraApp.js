// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

// Functional Components

const PrimeraApp = ({ greeting, subtitle }) => {
    return (
        <>
            <h1>{ greeting }</h1>
            {/* <pre>{ JSON.stringify(greeting, null, 3) }</pre> */}
            <p>{ subtitle }</p>
        </>
    ); 
}

export default PrimeraApp;

PrimeraApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle: 'I\'m a subtitle'
}