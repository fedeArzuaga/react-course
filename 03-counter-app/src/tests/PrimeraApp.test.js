import React from 'react'
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import PrimeraApp from "../PrimeraApp";

describe('PrimeraApp testing', () => {
    
    // test('should show the message "Salude caramba!"', () => {
        
    //     const greeting = "Salude caramba!";

    //     const { getByText } = render( <PrimeraApp greeting={greeting} /> );

    //     expect( getByText(greeting) ).toBeInTheDocument();

    // });

    test('should show <PrimeraApp /> correctly"', () => {
        
        const wrapper = shallow( <PrimeraApp greeting='Salude caramba!' /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should show the subtitle sent by props', () => {

        const greeting = "Salude caramba!";
        const wrapper = shallow(
            <PrimeraApp 
                greeting='Salude caramba!'
                subtitle="I'm a subtitle"
            />
        );

        const pText = wrapper.find('p').text();

        // console.log(pText);
        expect( pText ).toBe( "I'm a subtitle" );

    });

});