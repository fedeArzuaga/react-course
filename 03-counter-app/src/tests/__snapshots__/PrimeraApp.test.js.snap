// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp testing should show <PrimeraApp /> correctly" 1`] = `
<Fragment>
  <h1>
    Salude caramba!
  </h1>
  <p>
    I'm a subtitle
  </p>
</Fragment>
`;
